modules/javafx.web/src/main/java/com/sun/webkit/network/PublicSuffixes.java

```diff
@@ -57,10 +64,21 @@
 
 
     /**
-     * The mapping from domain names to public suffix list rules.
+     * The mapping from top-level domain names to public suffix list rules.
      */
-    private static final Map<String,Rule> RULES =
-            loadRules("effective_tld_names.dat");
+    private static final Map<String, Rules> rulesCache = new ConcurrentHashMap<>();
+
+
+    /*
+     * Determines whether the public suffix list file is available.
+     */
+    private static final boolean pslFileExists = AccessController.doPrivileged(
+        (PrivilegedAction<Boolean>) () -> {
+            String javaHome = System.getProperty("java.home");
+            String resourceName = "lib/security/public_suffix_list.dat";
+            File f = new File(javaHome, resourceName);
```

Comment:
I think it would be useful to log a warning here if the public suffix file does not exist (since this code is only executed once the warning would only be logged once).